crates/router-controller/src/controller.rs

```diff
@@ -127,8 +131,31 @@ impl Controller {
             0 => None,
             1 => Some(&candidates[0].url),
             len => {
-                let i: usize = rand::random();
-                Some(&candidates[i % len].url)
+                let key = (dataset.clone(), first_block);
+
+                let next_worker = if let Some(url) = self.request_tracker.get(&key) {
+                    let index = candidates.iter().position(|info| info.url == url);
+                    if let Some(index) = index {
+                        let worker = &candidates[index % len];
```

Comment:
`index + 1` ?

Reply:
yes. and i think i need to sort `candidates` at first